ext-dynamic-commitments.md

```diff
@@ -0,0 +1,1086 @@
+    - MUST NOT send another `dyn_propose`.
+    - SHOULD close the connection if it exceeds an acceptable time frame.
+  - if it sets `channel_type` and the `channel_type` conversion requires
+    re-anchoring (see appendix for conversions that require re-anchoring):
+    - MUST set `kickoff_feerate`.
+
+The receiving node:
+  - if `channel_id` does not match an existing channel it has with the sender:
+    - SHOULD send an `error` and close the connection.
+  - MUST respond with either a `dyn_ack` or `dyn_reject`.
+  - if the TLV parameters of the `dyn_propose` are acceptable and the receiver
+    intends to execute those parameter changes:
+    - MUST respond with `dyn_ack`.
+    - MUST remember its last received `dyn_propose` parameters.
+  - if the TLV parameters of the `dyn_propose` are NOT acceptable and the
+    receiver refuses to execute those parameter changes:
+    - MUST respond with `dyn_reject`.
+
+_NOTE FOR REVIEWERS_: These messages all interact with each other, so feedback
+is welcome for how to restructure this section so that the invariants it
+prescribes are found in the most intuitive place.
+
+##### Rationale
+
+The set of parameters used in this message to renegotiate channel parameters
+can't violate the invariants set out in BOLT 2. This is because we are simply
+trying change channel parameters without a close event. BOLT 2 specifies
+constraints on these parameters to make sure they are internally consistent and
+secure in all contexts.
+
+Since the `initiator` is the one that is responsible for paying the fees for the
+kickoff transaction if it is required (like for certain `channel_type` changes),
+it follows that the `responder` cannot change the `channel_type`. Since the
+`kickoff_feerate` paid by the `initiator`, it should be set only if the sender
+is the `initiator`.
+
+The requirement for a node to remember what it last _sent_ and for it to
+remember what it _accepted_ is necessary to recover on reestablish. See the
+reestablish section for more details.
+
+#### `dyn_ack`
+
+This message is sent in response to a `dyn_propose` indicating that it has
+accepted the proposal.
+
+1. type: 113 (`dyn_ack`)
+2. data:
+   * [`32*byte`:`channel_id`]
+
+##### Requirements
+
+The sending node:
+  - MUST set `channel_id` to a valid channel it has with the recipient.
+  - MUST NOT send this message if it has not received a `dyn_propose`
+  - MUST NOT send this message if it has already sent a `dyn_ack` for the
+    current negotiation.
+  - MUST NOT send this message if it has already sent a `dyn_reject` for the
+    current negotiation.
+  - MUST remember the parameters of `dyn_propose` message to which the `dyn_ack`
+    is responding for the next `dyn_height`.
+  - MUST remember the local and remote commitment heights for the next
+    `dyn_height`.
+
+The receiving node:
+  - if `channel_id` does not match an existing channel it has with the peer:
+    - MUST send an `error` and close the connection.
```

Comment:
Correct. I believe this should be `warning`. Closing the connection does not imply closing the channel, however, I do think that `error` has the semantic of signaling force close.

Reply:
Actually no this is definitely right. This is saying if we receive an unknown channel id we send an error and close the connection. We wouldn't fail the channel because the channel_id doesn't exist.